packages/rehype-plugin-codeblock/src/hast.ts

```diff
@@ -72,6 +73,60 @@ export function getHastElementType(
   };
 }
 
+export function wrapWithMetastringElements(
+  metastringInfo: MetastringInfo,
+  element: Element
+): Element[] {
+  const children: Element[] = [];
+
+  if (metastringInfo.isCollapsible) {
+    children.push({
+      type: 'element',
+      tagName: 'details',
+      children: [
+        {
+          type: 'element',
+          tagName: 'summary',
+          children: [
+            {
+              type: 'element',
+              tagName: 'span',
+              children: [
+                {
+                  type: 'text',
+                  value: metastringInfo.title
+                }
+              ]
+            }
+          ]
+        },
+        element
+      ]
+    });
+  } else if (metastringInfo.title) {
```

Comment:
Optional, add description on the structure that we push into children.

Reply:
Ah, yes. Sure! I can do that tomorrow. Thanks for the suggestion!

Reply:
@dio I added comments which represent the resulting JSX structure:

https://github.com/imballinst/protosaurus/blob/9fdc7572108501364c8d2e3e8e90425bcd8f3996/packages/rehype-plugin-codeblock/src/hast.ts#L84-L111

Let me know what you think! I'll merge this for now. If there are more things that you'd like to happen, I'll add it to my to-do.